### PR TITLE
Fix #14661 - Raw protobuf support

### LINKED ISSUE #14661: Add support for protobuf
protobuf is a self described serialization format that is widely used in applications and network communications. 

The offficial C impl is <4kloc and we can make it even smaller iff we just care about deserializing.

Example usage:

protoc --decode_raw < a

where 'a' contains "080c1022"

### PR DESCRIPTION
**Pull request by wargio**
Fix #14661 

image

### CODE REVIEW

**Comment by wargio** [comment]
https://github.com/radareorg/radare2-regressions/pull/2009

**Comment by radare** [comment]
can we have a json output? Looks awesome to me

File: libr/util/protobuf.c [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+#define WIRE_END_GRP   4 // groups (deprecated)
+#define WIRE_32_BIT    5 // fixed32, sfixed32, float
+
+typedef R_PACKED (struct _proto_head {
+	ut8 wire   : 3;
+	ut8 number : 5;
+}) proto_head_t;
+
+static const char* s_wire(const ut8 byte) {
+	switch(byte) {
```
Comment (radare) [inline libr/util/protobuf.c]: missing space

File: libr/util/protobuf.c [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+			decode_array(buffer - 1, end);
+			return;
+		}
+		buffer += bytes_read;
+	}
+}
+
+void r_protobuf_decode(const ut8* start, const ut64 size, bool debug) {
+	if (!start || !size) {
+		eprintf ("\ninvalid buffer pointer or size.\n");
```
Comment (radare) [inline libr/util/protobuf.c]: `s/\\ni/I/`. aka, dont start warning messages with a newline or lowercase

**Pull request merged**